codespell_lib/_codespell.py

```diff
@@ -29,6 +29,11 @@
 import textwrap
 
 word_regex_def = u"[\\w\\-'’`]+"
+# While we want to treat characters like ( or " as okay for a starting break,
+# these may occur unescaped in URIs, and so we are more restrictive on the
+# endpoint.  Emails are more restrictive, so the endpoint remains flexible.
+uri_regex_def = (u"(\\b(?:https?|t?ftp|file|git|smb)://[^\\s]+(?=$|\\s)|"
```

Comment:
What about `sftp`?
```suggestion
uri_regex_def = (u"(\\b(?:https?|[ts]?ftp|file|git|smb)://[^\\s]+(?=$|\\s)|"
```

Reply:
Seems reasonable, although it would be nice to sort the character set:
```suggestion
uri_regex_def = (u"(\\b(?:https?|[st]?ftp|file|git|smb)://[^\\s]+(?=$|\\s)|"
```